src/lib/Microsoft.Health.Common/EventHub/Listeners/EventHubListener.cs

```diff
@@ -230,6 +236,129 @@ private async Task TimedFlush()
                 }
             }
 
+            private async void TimeIntervalFlush()
+            {
+                while (!_cts.IsCancellationRequested)
+                {
+                    if (_currentIntervalEnd == DateTime.MinValue)
+                    {
+                        _currentIntervalEnd = CalculateIntervalEnd(_currentIntervalStart, _flushTimeSpan);
+                    }
+
+                    if (_currentIntervalEnd == DateTime.MinValue)

```

Comment:
Better to use a a reset event or semaphore to coordinate rather than a task delay.